src/Storages/MergeTree/ParallelReplicasReadingCoordinator.cpp

```diff
@@ -177,9 +178,17 @@ class ParallelReplicasReadingCoordinator::ImplInterface
     virtual ~ImplInterface() = default;
 
     virtual ParallelReadResponse handleRequest(ParallelReadRequest request) = 0;
-    virtual void handleInitialAllRangesAnnouncement(InitialAllRangesAnnouncement announcement) = 0;
+    virtual void doHandleInitialAllRangesAnnouncement(InitialAllRangesAnnouncement announcement) = 0;
     virtual void markReplicaAsUnavailable(size_t replica_number) = 0;
 
+    void handleInitialAllRangesAnnouncement(InitialAllRangesAnnouncement announcement)
+    {
+        if (++sent_initial_requests > replicas_count)
```

Comment:
this check looks silly, but the reality is we had(ve) a bug in joins with new analyzer that is recognisable exactly by this madness.